examples/deployment/scripts/transfer/gold_contract_sc.js

```diff
@@ -4,24 +4,36 @@
  * from contract account (contract approval mode) to another according to smart contract (ASC) logic
 */
 const { transferAlgo } = require('./common');
+const logicsig = require('algosdk/src/logicsig');
+const { decode } = require('@msgpack/msgpack');
 
 async function run(runtimeEnv, deployer) {
   const johnAccount = deployer.accountsByName.get("john-account");
   const elonMuskAccount = deployer.accountsByName.get("elon-musk-account");
 
-  // Transactions for Transaction for ALGO - Contract : '2-gold-contract-asc.teal'  (Contract Approval Mode)
-  const lsig = await deployer.getLogicSignature("2-gold-contract-asc.teal", []);
+  // Transactions for Transaction for ALGO - Contract : '2-gold-contract-asc.teal'  (Contract Mode)
+  //const lsig = await deployer.getLogicSignature("2-gold-contract-asc.teal", []);
   // sender is contract account
-  const sender = lsig.address(); 
-
+  const result = deployer.lsig.get("2-gold-contract-asc.teal").logicSignature;
+  const lsig = decode(result);
+  const dummyProgram = new Uint8Array([
+    1,  32,   2,   1, 100,  38,   1,  32, 213,   3, 149,
+   22,  62, 136, 178, 191, 199,  92, 201, 183, 175, 213,
+   79, 155, 211,  71, 149, 158, 180, 205, 247, 164, 218,
+  202,  87,  77, 111, 133, 236,  77,  49,  16,  34,  18,
+   49,   7,  40,  18,  16,  49,   8,  35,  14,  16
+]);
```

Comment:
this is only a dummyProgram , it will be overwritten later by retrieved logic signature.
- couln't leave it empty as constructor in `algosdk/src/logicsig` requires it

Reply:
OK, then you can have an empty array. All in all, we should hid all this details in the `deployer` function (`getDelegatedLsig`)

Reply:
OK?

Reply:
yes, i am trying to import `algosdk/src/logicsig` in deployer.
but getting error if i use `const logicsig = require('algosdk/src/logicsig')` as it has type any
i am working on a solution

Reply:
maybe you can move the code to `deployer`?